File: bubbleSort.py
class Solution:
    def bubbleSort(self, array):
        
        for count in range(len(array)-1):
            isSwapped= False

            for swi in range(len(array)-1-count):
                if array[swi] > array[swi+1]:
                    array[swi], array[swi+1] = array[swi+1], array[swi]
                    isSwapped = True
            
            if isSwapped == False:
                break

        
        return array



if __name__ == "__main__":
    obj = Solution()

    array = [9,8,7,6,5,4,3,2,1]

    print(obj.bubbleSort(array))

File: bubbleSort.java
public class bubbleSort {
    
    public int[] sorting(int array[]){

        int temp = 0;

        for (int swi=0; swi<array.length-1;swi++){
            for (int swj = 0; swj < array.length-swi-1;swj++){
                if(array[swj] > array[swj+1]){
                    temp = array[swj];
                    array[swj] = array[swj+1];
                    array[swj+1] = temp;
                }
            }
        }
        return array;
    }

    public static void main(String[] args){
    int[] array = {12, 3, 9, 32, 4, 56, 0, 1, 2 };

    bubbleSort obj = new bubbleSort();

    array = obj.sorting(array);
    System.out.print("Sorted Array: ");
    for(int swi=0;swi<array.length;swi++){
        System.out.print(array[swi] + " ");
    }
    }
}

// Simple Bubble Sort Problem in Java- Without any optimization